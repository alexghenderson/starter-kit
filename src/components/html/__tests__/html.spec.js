/**
 * @jest-environment node
 */

import enzyme from 'enzyme';
import Helmet from 'react-helmet';
import HTML from '../';

describe('HTML Component', () => {
  it('should render html document', () => {
    const scriptTag = '<script src="test"></script>';
    const helmet = Helmet.renderStatic();
    const wrapper = enzyme.shallow(
      <HTML helmet={helmet} scriptTags={scriptTag}>
        test-content
      </HTML>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  /**
   * @jest-environment jsdom
   */
  it('should throw if executed in browser', () => {
    process.env.BROWSER = true;
    const scriptTag = '<script src="test"></script>';
    const helmet = Helmet.renderStatic();
    expect(() => {
      enzyme.shallow(
        <HTML helmet={helmet} scriptTags={scriptTag}>
          test-content
        </HTML>
      );
    }).toThrow();
  });
});
