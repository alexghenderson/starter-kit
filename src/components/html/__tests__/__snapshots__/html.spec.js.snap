// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Component should render html document 1`] = `"<html><head><title data-react-helmet=\\"true\\"></title></head><body><div id=\\"app\\">test-content</div><script src=\\"test\\"></script></body></html>"`;
